cni/pkg/nodeagent/server.go

```diff
@@ -185,9 +186,23 @@ func (s *Server) Start() {
 	s.handlers.Start()
 }
 
-func (s *Server) Stop() {
+func (s *Server) Stop(skipCleanup bool) {
 	s.cniServerStopFunc()
-	s.dataplane.Stop()
+	s.dataplane.Stop(skipCleanup)
+}
+
+func (s *Server) ShouldStopForUpgrade(selfName, selfNamespace string) bool {
+	dsName := fmt.Sprintf("%s-node", selfName)
```

Comment:
Note to myself -- the helm template hardcodes exactly the same, so this checks out :+1:

Reply:
Yeah I actually don't really love this and may improve but at least the integ test added here will fail if these get out of sync.